### PR TITLE
Internal: Repeater - Implement min_items functionality

### PR DESCRIPTION
**Pull request by nicoladj77**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Add the ability to set a min number of items for a repeater

## Description
An explanation of what is done in this PR

* Add the ability to set a min number of items for a repeater

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -39,3 +39,6 @@ tests/playwright/blob-report
 **/allure-results
 playwright-test-results
 **/.env
+
+# phpunit
+.phpunit.result.cache
```
Comment (louiswol94) [inline .gitignore:44]: Please revert this change

File: assets/dev/js/editor/controls/repeater.js [hunk 2]
```diff
@@ -113,11 +113,14 @@ ControlRepeaterItemView = ControlBaseDataView.extend( {
 	},
 
 	toggleMinRowsClass() {
-		if ( ! this.model.get( 'prevent_empty' ) ) {
-			return;
+		let minItems = 0;
+		if ( this.model.get( 'min_items' ) && Number.isInteger( this.model.get( 'min_items' ) ) ) {
+			minItems = this.model.get( 'min_items' );
+		} else if ( this.model.get( 'prevent_empty' ) ) {
+			minItems = 1;
```
Comment (louiswol94) [inline assets/dev/js/editor/controls/repeater.js:120]: Extract to `getMinItems()`

File: includes/controls/repeater.php [hunk 3]
```diff
@@ -181,4 +182,20 @@ public function content_template() {
...
+				count( $control_data['default'] ) < $control_data['min_items']
+			) {
+				throw new \Exception(
+					__( 'In a Repeater control, if you specify a minimum number of items, you must also specify a default value that contains at least that number of items.', 'elementor' )
+				);
+			}
+		}
+
+		return true;
+	}
```
Comment (louiswol94) [inline includes/controls/repeater.php:200]: @nicoladj77 I am wondering if we should do this validation client side?

See: `elementor/assets/dev/js/editor/components/validator/number.js` as an example. This might not work specific for our scenario, but maybe just follow the same logic.

I am just wondering if it is dangerous to throw this kind of Exception that would totally break the Editor if a 3rd party does something wrong.

**Review by louiswol94 — CHANGES_REQUESTED** [review]

File: includes/controls/repeater.php (hunk 3 above)
Comment (nicoladj77) [reply includes/controls/repeater.php:200]: @louiswol94 this exception should only be thrown in the context of the Controls_Manager

		if ( $control_type_instance instanceof Has_Validation ) {
			try {
				$control_type_instance->validate( $control_data );
			} catch ( \Exception $e ) {
				_doing_it_wrong( sprintf( '%1$s::%2$s', __CLASS__, __FUNCTION__ ), esc_html( $e->getMessage() ), '3.23.0' );
				return false;
			}
		}

I totally understand your concern, I can of course to this differently, returning an array from validate() with two keys, valid and message. I wanted to run this by you because like this is cleaner from a coding perspective, but if you think it's not safe, let's do it the other way.

Comment (nicoladj77) [reply includes/controls/repeater.php:200]: @louiswol94 regarding FE vs BE, this to me looks similar to

		$control_type_instance = $this->get_control( $control_data['type'] );

		if ( ! $control_type_instance ) {
			_doing_it_wrong( sprintf( '%1$s::%2$s', __CLASS__, __FUNCTION__ ), sprintf( 'Control type "%s" not found.', esc_html( $control_data['type'] ) ), '1.0.0' );
			return false;
		}

which means that if a control is in a not valid state, it should not be added. But let me know if you think differently. if I had to do the check in the FE, I would add rows until they are reach the min_items value.

**Review by louiswol94 — APPROVED** [review]
